tests/johnson_alahi_li_2016/test_modules.py

```diff
@@ -0,0 +1,315 @@
+                    in_out_channels.append(
+                        (
+                            module.residual[0][0].in_channels,
+                            module.residual[-1][0].out_channels,
+                        )
+                    )
+
+        with subtests.test("channel_config"):
+            assert in_out_channels == channel_config
+
+
+def test_johnson_alahi_li_2016_transformer_decoder(subtests):
+    channel_configs = [[(64, 32), (32, 16), (16, 3)], [(128, 64), (64, 32), (32, 3)]]
+
+    for instance_norm, channel_config in zip((True, False), channel_configs):
+        with subtests.test(instance_norm=instance_norm):
+            decoder = modules.johnson_alahi_li_2016_transformer_decoder(
+                instance_norm=instance_norm
+            )
+
+            assert isinstance(type(decoder), type(pystiche.SequentialModule))
+
+            in_out_channels = []
+            for i, module in enumerate(decoder.children()):
+                if i in range(2):
+                    with subtests.test("conv_layer"):
+                        assert isinstance(type(module), type(nn.Sequential))
+                        in_out_channels.append(
+                            (module[0].in_channels, module[0].out_channels)
+                        )
+                if i == 2:
+                    with subtests.test("output_conv"):
+                        assert isinstance(type(module), type(nn.Conv2d))
+                        in_out_channels.append(
+                            (module.in_channels, module.out_channels)
+                        )
+
+            with subtests.test("channel_config"):
+                assert in_out_channels == channel_config
+
+
+def test_johnson_alahi_li_2016_transformer_decoder_value_range_delimiter(
+    subtests, input_image
+):
+    for impl_params in (True, False):
+        with subtests.test(impl_params=impl_params):
+            decoder = modules.johnson_alahi_li_2016_transformer_decoder(
+                impl_params=impl_params
+            )
+
+            module = [x for x in decoder.children()][-1]
+            assert isinstance(type(module), type(nn.Module))
+
+            with subtests.test("delimiter"):
+                actual = module(input_image)
+                desired = (
+                    150.0 * torch.tanh(input_image)
+                    if impl_params
+                    else torch.sigmoid(2.0 * input_image)
+                )
+
+                ptu.assert_allclose(actual, desired)
+
+
+def test_JohnsonAlahiLi2016Transformer(image_medium):
+    transformer = modules.JohnsonAlahiLi2016Transformer()
+
+    assert isinstance(type(transformer.encoder), type(pystiche.SequentialModule))
+    assert isinstance(type(transformer.decoder), type(pystiche.SequentialModule))
+
+    output_image = transformer(image_medium)
+    assert image_medium.size() == output_image.size()
+
+
+def test_johnson_alahi_li_2016_transformer():
```

Comment:
We need to test two more things:
1. Can a transformer be loaded for all available weights?
2. Is the loaded `state_dict` correct?

I'll take this up.